apps/full-site-editing/full-site-editing-plugin/blocks/content-renderer/render.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+function render_content_renderer_block( $attributes ) {
+	if ( ! $attributes['selectedPostId'] || ! is_int( $attributes['selectedPostId'] ) ) {
+		return;
+	}
+	$align = $attributes['align'] ? ' align' . $attributes['selectedPostId'] : '';
+	$post = get_post( $attributes['selectedPostId'] );
+	setup_postdata( $post );
+	$content = '<div class="a8c-content-renderer'. $align . '">' . get_the_content() . '</div>';
```

Comment:
Need to do `apply_filters( 'the_content', get_the_content() )` so we get `do_blocks` and other expected formatting. Try rendering a post with a dynamic block (eg Latest Posts) in the post you're rendering and it won't render right now.

Reply:
TIL!

Reply:
Damn this has also fixed an error I was struggling to figure out.

Post X contains a renderer rendering post Y which contains another renderer for post Z.
On the front-end, post X shows Y correctly, but not Z.
This was because I wasn't `do_blocks`-ing Y!

TWISTED! 🧶

Reply:
To extend this a bit further, we can also create circular dependencies now. For example: Post X contains a renderer for Post Y, and Post Y contains a rendered for Post X. When you try to save this the updating will fail, and if you try to reload the editor it will get into an infinite loop (`Allowed memory size of 268435456 bytes exhausted (tried to allocate 262144 bytes) in /var/www/html/wp-includes/plugin.php on line 465`)

Anyway, this is and edge case that we can tackle in follow-up PRs. 😄